rcl/src/rcl/wait.c

```diff
@@ -556,14 +564,6 @@ rcl_wait(rcl_wait_set_t * wait_set, int64_t timeout)
         wait_set->timers[i] = NULL;
         continue;
       }
-      rmw_guard_conditions_t * rmw_gcs = &(wait_set->impl->rmw_guard_conditions);
-      size_t gc_idx = wait_set->size_of_guard_conditions + i;
-      if (NULL != rmw_gcs->guard_conditions[gc_idx]) {
-        // This timer has a guard condition, so move it to make a legal wait set.
-        rmw_gcs->guard_conditions[rmw_gcs->guard_condition_count] =
-          rmw_gcs->guard_conditions[gc_idx];
-        ++(rmw_gcs->guard_condition_count);
-      }
```

Comment:
Why was this section moved?

Reply:
Although timer is canceled, timer gc must add to waitset gc in order to trigger gc when canceled timer is reset.
On Above code, if timer is canceled, skip this code. So, I move this section above code for checking cancellation of timer.

if (is_canceled) {
    wait_set->timers[i] = NULL;
    continue;
}

Reply:
Although timer is canceled, timer gc must add to waitset gc in order to trigger gc when canceled timer is reset.
On Above code, if timer is canceled, skip this code. So, I move this section above code for checking cancellation of timer.
```cpp
if (is_canceled) {
    wait_set->timers[i] = NULL;
    continue;
}
```

Reply:
Ok, sounds good. 👍